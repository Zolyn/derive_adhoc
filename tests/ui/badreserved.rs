// Tests of syntax we may wish to use for things in the future

use derive_adhoc::{define_derive_adhoc, derive_adhoc, Adhoc};

#[derive(Adhoc)]
struct ReserveDriver;

derive_adhoc! {
    ReserveDriver:
    $[ future ]
}

derive_adhoc! {
    ReserveDriver:
    $< paste? >
}

derive_adhoc! {
    ReserveDriver:
    $r#"template content?"#
}

derive_adhoc! {
    ReserveDriver:

    mod prevent {
        //! template doc comment?
    }
}

derive_adhoc! {
    ReserveDriver:

    $/// template doc comment?
    mod prevent {
    }
}

fn main() {}
