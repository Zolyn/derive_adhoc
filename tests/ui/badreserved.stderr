error: expected one of: `$`, parentheses, curly braces, identifier
  --> ui/badreserved.rs:10:6
   |
10 |     $[ future ]
   |      ^

error: expected one of: `$`, parentheses, curly braces, identifier
  --> ui/badreserved.rs:15:6
   |
15 |     $<ident:upper>
   |      ^

error: not allowed in within ${paste ...} (or case_changing)
  --> ui/badreserved.rs:16:18
   |
16 |     ${paste ident:upper}
   |                  ^

error: expected one of: `$`, parentheses, curly braces, identifier
  --> ui/badreserved.rs:21:6
   |
21 |     $r#"template content?"#
   |      ^^^^^^^^^^^^^^^^^^^^^^

error: inner attributes are reserved syntax, anywhere in derive-adhoc templates
  --> ui/badreserved.rs:28:9
   |
28 |         //! template doc comment?
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected one of: `$`, parentheses, curly braces, identifier
  --> ui/badreserved.rs:35:6
   |
35 |     $/// template doc comment?
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `define_derive_adhoc`
 --> ui/badreserved.rs:3:20
  |
3 | use derive_adhoc::{define_derive_adhoc, derive_adhoc, Adhoc};
  |                    ^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
